### PR TITLE
reference/configuration: update tikv configuration file

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed, added or deleted? (Required)

- Add shared block cache related configs
- Add Titan newly added configs

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/configuration/tikv-server/configuration-file.md [hunk 1]
```diff
@@ -1022,6 +1036,22 @@ rocksdb defaultcf titan 相关的配置项。
 + 最小值：0
 + 单位：KB|MB|GB
 
+### `blob-run-mode`
+
++ Titan 的运行模式选择：正常写入；只读不再写入 blob；回退将 blob 写回 LSM
```
Comment (zhangjinpeng1987) [inline reference/configuration/tikv-server/configuration-file.md]: s/正常写入/value size 超过 min-blob-size 的数据会写入到 blob 文件
s/只读不再写入 blob/不再写入新数据到 blob，原有 blob 内的数据仍然可以读取
s/回退将 blob 写回 LSM/将 blob 内的数据写回 LSM

File: reference/configuration/tikv-server/configuration-file.md [hunk 2]
```diff
@@ -1022,6 +1036,22 @@ rocksdb defaultcf titan 相关的配置项。
...
++ 默认值："normal"
++ 可选值："normal"|"read_only"|"fallback"
+
+### `level-merge`
+
++ 是否开启 level-merge 牺牲些写放大以提升读性能
```
Comment (zhangjinpeng1987) [inline reference/configuration/tikv-server/configuration-file.md]: s/是否通过开启 level-merge 来提升读性能，副作用是写放大会比不开启更大

File: reference/configuration/tikv-server/configuration-file.md [hunk 3]
```diff
@@ -1022,6 +1036,22 @@ rocksdb defaultcf titan 相关的配置项。
...
++ 默认值：true
+
+### `gc-merge-rewrite`
+
++ 是否开启使用 merge operator 来进行 Titan GC 写回
```
Comment (zhangjinpeng1987) [inline reference/configuration/tikv-server/configuration-file.md]: s/是否开启使用 merge operator 来进行 Titan GC 写回操作，减少 Titan GC 对于前台写入的影响

File: reference/configuration/tikv-server/configuration-file.md [hunk 4]
```diff
@@ -257,6 +257,20 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
 + 默认值：2GB
 + 单位: MB|GB
 
+## storage.block-cache
+
+Rocksdb 多个 CF 之间共享 block cache 相关的配置选项，当开启时，为每个 CF 单独配置的 block cache 将无效。
```
Comment (zhangjinpeng1987) [inline reference/configuration/tikv-server/configuration-file.md]: s/相关的配置选项/的配置选项

File: reference/configuration/tikv-server/configuration-file.md [hunk 5]
```diff
@@ -257,6 +257,20 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
...
+
+### `shared`
+
++ 是否开启共享 block cache
++ 默认值：true
+
+###  `capacity`
+
++ 共享 block cache 的大小
++ 默认值：系统总内存大小的 45%
```
Comment (zhangjinpeng1987) [inline reference/configuration/tikv-server/configuration-file.md]: 缺少单位说明

File: reference/configuration/tikv-server/configuration-file.md [hunk 6]
```diff
@@ -257,6 +257,21 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
 + 默认值：2GB
 + 单位: MB|GB
 
+## storage.block-cache
+
+Rocksdb 多个 CF 之间共享 block cache 的配置选项，当开启时，为每个 CF 单独配置的 block cache 将无效。
```
Comment (lilin90) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
RocksDB 多个 CF 之间共享 block cache 的配置选项，当开启时，为每个 CF 单独配置的 block cache 将无效。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 7]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
 + 最小值：0
 + 单位：KB|MB|GB
 
+### `blob-run-mode`
+
++ Titan 的运行模式选择，可选值：
+    + "normal"：value size 超过 min-blob-size 的数据会写入到 blob 文件
+    + "read_only": 不再写入新数据到 blob，原有 blob 内的数据仍然可以读取
```
Comment (lilin90) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
    + "read_only"：不再写入新数据到 blob，原有 blob 内的数据仍然可以读取
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 8]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
...
+    + "fallback": 将 blob 内的数据写回 LSM
```
Comment (lilin90) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
    + "fallback"：将 blob 内的数据写回 LSM
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 9]
```diff
@@ -257,6 +257,21 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
 + 默认值：2GB
 + 单位: MB|GB
 
+## storage.block-cache
+
+RocksDB 多个 CF 之间共享 block cache 的配置选项，当开启时，为每个 CF 单独配置的 block cache 将无效。
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]: 是否有默认值呢？
```suggestion
RocksDB 多个 CF 之间共享 block cache 的配置选项。当开启时，为每个 CF 单独配置的 block cache 将无效。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 10]
```diff
@@ -257,6 +257,21 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
...
+
+### `shared`
+
++ 是否开启共享 block cache
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 控制是否开启共享 block cache。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 11]
```diff
@@ -257,6 +257,21 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
...
++ 默认值：true
+
+### `capacity`
+
++ 共享 block cache 的大小
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 共享 block cache 的大小。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 12]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
...
+
++ Titan 的运行模式选择，可选值：
+    + "normal"：value size 超过 min-blob-size 的数据会写入到 blob 文件
+    + "read_only"：不再写入新数据到 blob，原有 blob 内的数据仍然可以读取
+    + "fallback"：将 blob 内的数据写回 LSM
++ 默认值："normal"
+
+### `level-merge`
+
++ 是否通过开启 level-merge 来提升读性能，副作用是写放大会比不开启更大
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]: level-merge 不就是本配置项吗？
```suggestion
+ 决定是否通过开启 level-merge 来提升读性能，副作用是写放大会比不开启本项更大。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 13]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
...
++ 默认值：true
+
+### `gc-merge-rewrite`
+
++ 是否开启使用 merge operator 来进行 Titan GC 写回操作，减少 Titan GC 对于前台写入的影响
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 控制是否开启使用 merge operator 来进行 Titan GC 写回操作，减少 Titan GC 对于前台写入的影响。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 14]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
 + 最小值：0
 + 单位：KB|MB|GB
 
+### `blob-run-mode`
+
++ Titan 的运行模式选择，可选值：
+    + "normal"：value size 超过 min-blob-size 的数据会写入到 blob 文件
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]: 请为本行、L1044 和 L1045 行加上句号。

File: reference/configuration/tikv-server/configuration-file.md (hunk 10 above)
Comment (Connor1996) [reply reference/configuration/tikv-server/configuration-file.md]: 我看这个文档里的其他类似描述都是直接用的`是否`

File: reference/configuration/tikv-server/configuration-file.md (hunk 14 above)
Comment (Connor1996) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
    + "normal"：value size 超过 min-blob-size 的数据会写入到 blob 文件。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 15]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
...
+    + "read_only"：不再写入新数据到 blob，原有 blob 内的数据仍然可以读取
```
Comment (Connor1996) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
    + "read_only"：不再写入新数据到 blob，原有 blob 内的数据仍然可以读取。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 16]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
...
+    + "fallback"：将 blob 内的数据写回 LSM
```
Comment (Connor1996) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
    + "fallback"：将 blob 内的数据写回 LSM。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 17]
```diff
@@ -257,6 +257,21 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
 + 默认值：2GB
 + 单位: MB|GB
 
+## storage.block-cache
+
+RocksDB 多个 CF 之间共享 block cache 的配置选项。当开启时，为每个 CF 单独配置的 block cache 将无效。
+
+### `shared`
+
++ 是否开启共享 block cache
```
Comment (Connor1996) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 是否开启共享 block cache。
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 18]
```diff
@@ -1022,6 +1037,24 @@ rocksdb defaultcf titan 相关的配置项。
...
+
++ Titan 的运行模式选择，可选值：
+    + "normal"：value size 超过 min-blob-size 的数据会写入到 blob 文件。
+    + "read_only"：不再写入新数据到 blob，原有 blob 内的数据仍然可以读取。
+    + "fallback"：将 blob 内的数据写回 LSM。
++ 默认值："normal"
+
+### `level-merge`
+
++ 是否通过开启 level-merge 来提升读性能，副作用是写放大会比不开启更大
```
Comment (Connor1996) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ 是否通过开启 level-merge 来提升读性能，副作用是写放大会比不开启更大。
```

**Comment by Connor1996** [comment]
PTAL @TomShawn @lilin90

**Pull request merged**